src/network/mcpe/handler/ItemStackRequestExecutor.php

```diff
@@ -333,11 +335,20 @@ protected function processItemStackRequestAction(ItemStackRequestAction $action)
 
 			$this->setNextCreatedItem($item, true);
 		}elseif($action instanceof CraftRecipeStackRequestAction){
-			$this->beginCrafting($action->getRecipeId(), 1);
+			$recipeId = $action->getRecipeId();
+			$window = $this->player->getCurrentWindow();
+			if($window instanceof EnchantInventory){
+				$this->specialTransaction = new EnchantTransaction($this->player, $recipeId);
+				$this->setNextCreatedItem($window->getOutput($recipeId));
+				return;
```

Comment:
early return should not be used in complex code like this - an if/else should be preferred